test/new-e2e/tests/installer/unix/package_agent_test.go

```diff
@@ -422,14 +422,15 @@ func (s *packageAgentSuite) TestUpgrade_DisabledAgentDebRPM_to_OCI() {
 	state = s.host.State()
 	s.assertUnits(state, false)
 	s.host.AssertPackageInstalledByInstaller("datadog-agent")
-	s.host.AssertPackageInstalledByPackageManager("datadog-agent")
+	s.host.AssertPackageNotInstalledByPackageManager("datadog-agent")
 
 	s.host.Run("sudo systemctl show datadog-agent -p ExecStart | grep /opt/datadog-packages")
 }
 
 func (s *packageAgentSuite) TestInstallWithLeftoverDebDir() {
 	// create /opt/datadog-agent to simulate a disabled agent
 	s.host.Run("sudo mkdir -p /opt/datadog-agent")
+	defer func() { s.host.Run("sudo rm -rf /opt/datadog-agent") }()
```

Comment:
Why is this needed ?

Reply:
The test doesn't clean up properly if it fails and fails the next tests. I had issues debugging because of it.